File: bench/repos/mini_ts_query_string_service/tests/public_checks.ts
import assert from "node:assert/strict";

import { canonicalQuery } from "../query.ts";

assert.equal(canonicalQuery({ q: "alpha beta", page: 2 }), "page=2&q=alpha%20beta");
assert.equal(canonicalQuery({ tag: ["pro", "beta"] }), "tag=pro&tag=beta");

File: bench/repos/mini_py_query_string_service/tests/public_checks.py
from pathlib import Path
import sys

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from query import canonical_query


assert canonical_query({"q": "alpha beta", "page": 2}) == "page=2&q=alpha+beta"
assert canonical_query({"tag": ["pro", "beta"]}) == "tag=pro&tag=beta"
